src/mailmap.rs

```diff
@@ -0,0 +1,215 @@
+      MailmapInner::Owned(owned) => owned.resolve_signature(&git_signature)?,
+    };
+
+    Signature::try_from(resolved)
+  }
+}
+
+#[napi]
+impl Commit {
+  #[napi]
+  /// Get the author of this commit, using the mailmap to map it to the canonical name and email.
+  ///
+  /// @category Commit/Methods
+  ///
+  /// @signature
+  /// ```ts
+  /// class Commit {
+  ///   authorWithMailmap(mailmap: Mailmap): Signature;
+  /// }
+  /// ```
+  ///
+  /// @param {Mailmap} mailmap - The mailmap to use for mapping
+  /// @returns Author signature of this commit with mapping applied
+  pub fn author_with_mailmap(&self, mailmap: &Mailmap) -> crate::Result<Signature> {
+    let git_signature = match &mailmap.inner {
+      MailmapInner::Repo(repo) => self.inner.author_with_mailmap(repo)?,
+      MailmapInner::Owned(owned) => self.inner.author_with_mailmap(owned)?,
+    };
+    let signature = Signature::try_from(git_signature)?;
+    Ok(signature)
+  }
+
+  #[napi]
+  /// Get the committer of this commit, using the mailmap to map it to the canonical name and email.
+  ///
+  /// @category Commit/Methods
+  ///
+  /// @signature
+  /// ```ts
+  /// class Commit {
+  ///   committerWithMailmap(mailmap: Mailmap): Signature;
+  /// }
+  /// ```
+  ///
+  /// @param {Mailmap} mailmap - The mailmap to use for mapping
+  /// @returns Committer signature of this commit with mapping applied
+  pub fn committer_with_mailmap(&self, mailmap: &Mailmap) -> crate::Result<Signature> {
+    let git_signature = match &mailmap.inner {
+      MailmapInner::Repo(repo) => self.inner.committer_with_mailmap(repo)?,
+      MailmapInner::Owned(owned) => self.inner.committer_with_mailmap(owned)?,
+    };
+    let signature = Signature::try_from(git_signature)?;
+    Ok(signature)
+  }
+}
+
+#[napi]
+impl Repository {
+  /// Gets this repository's mailmap.
+  ///
+  /// @category Repository/Methods
+  ///
+  /// @signature
+  /// ```ts
+  /// class Repository {
+  ///   mailmap(): Mailmap | null;
+  /// }
+  /// ```
+  ///
+  /// @returns The mailmap object if it exists, null otherwise
+  #[napi]
+  pub fn mailmap(&self, this: Reference<Repository>, env: Env) -> Option<Mailmap> {
+    let inner = this
+      .share_with(env, |repo| {
+        repo.inner.mailmap().map_err(crate::Error::from).map_err(|e| e.into())
+      })
+      .ok()?;
+
+    Some(Mailmap {
+      inner: MailmapInner::Repo(inner),
+    })
+  }
+}
+
+/// Create a mailmap from the contents of a string.
+///
+/// The format of the string should follow the rules of the mailmap file:
+/// ```
+/// # Comment line (ignored)
+/// Seokju Me <seokju.me@toss.im> Seokju Na <seokju.me@gmail.com>
+/// ```
+///
+/// @param {string} content - Content of the mailmap file
+/// @returns A new mailmap object
+/// @throws An error if operation failed
+///
+/// @category Mailmap/Methods
+///
+/// @signature
+/// ```ts
+/// static createMailmapFromBuffer(content: string): Mailmap;
```

Comment:
```suggestion
/// @category Mailmap
///
/// @signature
/// ```ts
/// function createMailmapFromBuffer(content: string): Mailmap;
```
I would like to suggest some changes to jsdoc.